python/sglang/srt/utils/common.py

```diff
@@ -1910,27 +1911,60 @@ def get_npu_compiler_config():
     return config
 
 
-def get_compiler_backend() -> str:
+def get_compiler_backend(
+    mode: str = None,
+    model_runner=None,
+    compilation_config: CompilationConfig = None,
+    compilation_context=None,
+) -> str:
     if hasattr(torch, "hpu") and torch.hpu.is_available():
         return "hpu_backend"
 
     if hasattr(torch, "npu") and torch.npu.is_available():
-        try:
-            import torchair
-            import torchair.ge_concrete_graph.ge_converter.experimental.patch_for_hcom_allreduce
-            from torchair.configs.compiler_config import CompilerConfig
-        except ImportError as e:
-            raise ImportError(
-                "NPU detected, but torchair package is not installed. "
-                "Please install torchair for torch.compile support on NPU."
+        if compilation_config is None:
+            compilation_config = CompilationConfig(compiler="torchair")
+
+        if compilation_config.compiler == "piecewise":
+            from sglang.srt.hardware_backend.npu.graph_runner.compilation.piecewise_npu_graph_compiler_backend import (
+                PiecewiseNpuGraphCompilerBackend,
             )
-        compiler_config = CompilerConfig()
-        predefined_config = get_npu_compiler_config()
-        for k, v in predefined_config.items():
-            setattr(compiler_config.experimental_config, k, v)
 
-        npu_backend = torchair.get_npu_backend(compiler_config=compiler_config)
-        return npu_backend
+            return PiecewiseNpuGraphCompilerBackend(
+                model_runner, compilation_config, compilation_context
+            )
+
+        if compilation_config.compiler == "npugraph":
+            from sglang.srt.hardware_backend.npu.graph_runner.compilation.npu_graph_compiler_backend import (
+                NpuGraphCompilerBackend,
+            )
+
+            return NpuGraphCompilerBackend(model_runner)
+
+        if compilation_config.compiler == "torchair":
+            try:
+                import torchair
```

Comment:
It's not a best practice to import modules inside a function unless being inevitable, needless to say in such a critical path.